tests/core/test_helpers.py

```diff
@@ -0,0 +1,26 @@
+import platform
+import site
+from pathlib import Path
+
+import pytest
+
+from singer_sdk.helpers._compat import get_project_distribution, metadata
+
+
+def test_get_project_distribution():
+    """Test `get_project_distribution()`.
+
+    click is a representative example of a distributed package from our dependency tree.
+    Any similar singer_sdk dependency could be used in stead.
+    """
+    if platform.system() == "Windows":
+        pytest.xfail("Doesn't pass on windows.")
```

Comment:
Does this mean the package name can't be retrieved in Windows?

Reply:
It means the lookup via distribution will not work, yes. The fallback to `metadata.version(cls.name)` was the primary mechanism before this change (it relies on the library name being the same as the class name), though I don't know how well it worked for Windows in the first place 🤔

Given the hurdles we have to jump through to try and infer the plugin version via distribution or library metadata, I still  think we are better off adding `__version__ = "<version>"` in `__init__.py` and importing that for `--about`. `poetry-dynamic-versioning` supports substitution from a `0.0.0` placeholder in any file (including `__init__.py` by default), and this approach is standard in other python packages 🤷‍♂️ This PR specifically addresses [this ask](https://github.com/meltano/sdk/pull/1241#pullrequestreview-1207289029) from @aaronsteers 👍

Reply:
@kgpayne Yeah, I'd say it's probably not worth spending time trying to solve a problem that's only present when migrating existing taps and targets that are published to PyPI. Wdyt?